utils/src/ooinstall/oo_config.py

```diff
@@ -38,23 +38,14 @@ def __init__(self, **kwargs):
         self.public_hostname = kwargs.get('public_hostname', None)
         self.connect_to = kwargs.get('connect_to', None)
         self.preconfigured = kwargs.get('preconfigured', None)
-
-        # Should this host run as an OpenShift master:
-        self.master = kwargs.get('master', False)
-
-        # Should this host run as an OpenShift node:
-        self.node = kwargs.get('node', False)
-
-        # Should this host run as an HAProxy:
-        self.master_lb = kwargs.get('master_lb', False)
-
         self.containerized = kwargs.get('containerized', False)
+        self.roles = kwargs.get('roles', None)
 
         if self.connect_to is None:
-            raise OOConfigInvalidHostError("You must specify either an ip " \
-                "or hostname as 'connect_to'")
+            raise OOConfigInvalidHostError(
+                "You must specify either an ip or hostname as 'connect_to'.")
 
-        if self.master is False and self.node is False and self.master_lb is False:
+        if self.is_master is False and self.is_node is False and self.is_master_lb is False:
             raise OOConfigInvalidHostError(
                 "You must specify each host as either a master or a node.")
```

Comment:
Error message does not match the condition.  Also, are we assuming that is_master_lb will also host storage as well? if so, we should probably rename it.